app/services/migration/migrators/user.rb

```diff
@@ -32,17 +32,18 @@ def records_per_worker
 
     def call
       migrate(self.class.ecf_users) do |ecf_user|
-        user = ::Migration::Users::Creator.new(ecf_user).find_or_initialize
+        npq_application = most_recent_created_npq_application(ecf_user)
+        email = npq_application&.participant_identity&.email
+        email = email.downcase if email.present?
```

Comment:
curious when there would be no email 🤔 maybe we need to fallback to profiles if there are no applications incase to avoid a case where the email is `nil`?

Reply:
No email means there's no npq_application, which wouldn't be included in the migration.
But either way, the `ecf_id` and `gai_id` would resolve to a user.

Reply:
the thing is we have cases where the user is only connected the profile not the application so this would be blank, but that is just bad data we should resolve